src/app/lib/controller.js

```diff
@@ -95,7 +94,8 @@ class Controller {
         'outline:start': [],
         'file:load': [],
         'file:unload': [],
-        'homing:flag': []
+        'homing:flag': [],
+        'electron-error': [],
```

Comment:
There is already an error listener setup, I'm not sure the reason we'd need to add another error listener.

You can probably refactor this code to use the 'error' event listener and the 'type' key can differentiate between what sort of error it is (firmware related, app related, etc.) - and remove this unique handler.